test/e2e_tsa_test.go

```diff
@@ -175,22 +177,29 @@ func generateMTLSKeys(t *testing.T, td string) (string, string, string, string,
 	return pemRootRef, serverPemLeafRef, serverPemKeyRef, clientPemLeafRef, clientPemKeyRef
 }
 
-func setUpTSAServerWithTLS(t *testing.T, td, timestampCACert, timestampServerKey, timestampServerCert string) (string, string, func()) {
+func setUpTSA(t *testing.T, td, timestampCACert, timestampServerKey, timestampServerCert string) (string, string, func()) {
 	viper.Set("timestamp-signer", "memory")
 	viper.Set("timestamp-signer-hash", "sha256")
 	viper.Set("disable-ntp-monitoring", true)
-	viper.Set("tls-host", "0.0.0.0")
-	viper.Set("tls-port", 3000)
-	viper.Set("tls-ca", timestampCACert)
-	viper.Set("tls-key", timestampServerKey)
-	viper.Set("tls-certificate", timestampServerCert)
-	tsaAPIServer := tsaserver.NewRestAPIServer("localhost", 3000, []string{"https"}, false, 10*time.Second, 10*time.Second)
-	tsaServer := httptest.NewServer(tsaAPIServer.GetHandler())
-	tsaClient, err := tsaclient.GetTimestampClient(tsaServer.URL)
+	close := func() {}
+	tsaURLPrefix := os.Getenv("TSA_URL")
+	if tsaURLPrefix == "" { // use tsa as a library instead.
```

Comment:
Do we need this or can we simplify and assume the TSA is running?

Reply:
No, because currently the Action doesn't work on macOS, and these tests are intended to be cross-os

 - https://github.com/sigstore/cosign/pull/4161/files/1ff5abf60eccda2cca517ab65e56a9342ec0b497

Reply:
Can this be tracked as a future improvement then? Would be nice to clean this up.

Reply:
tracking in https://github.com/sigstore/scaffolding/issues/1555